website/components/_app.tsx

```diff
@@ -61,6 +64,50 @@ const DefaultSeoWithLocale = () => {
 
 const MyApp = ({ Component, pageProps, router }: AppProps) => {
   const localeSwitcher = useMemo(() => <LocaleSwitcher key="localeSwitcher" />, [])
+  // @ts-expect-error todo: fix type error
+  const __nextra_internal__ = (globalThis as NextraInternalGlobal)[NEXTRA_INTERNAL]
+
+  const internal = __nextra_internal__.context?.[router.route]
+  const hasMDXPage = !!internal
+  const { pageOpts = {} } = internal || {}
+  const locale = router.asPath.split('/')[1] as Locale
+
+  pageOpts.headings = useMemo(() => {
+    if (!hasMDXPage) return []
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return pageOpts.filePath === 'pages/[locale]/index.mdx'
+      ? [
+          {
+            id: 'network-roles',
+            value: translate(translations, locale, 'index.networkRoles.title'),
+            depth: 2,
+          },
+          {
+            id: 'products',
+            value: translate(translations, locale, 'index.products.title'),
+            depth: 2,
+          },
+          {
+            id: 'supported-networks',
+            value: translate(translations, locale, 'index.supportedNetworks.title'),
+            depth: 2,
+          },
+        ]
+      : pageOpts.headings
+  }, [hasMDXPage, pageOpts.filePath, pageOpts.headings, locale])
+
+  pageOpts.pageMap = useMemo(() => {
+    if (!hasMDXPage) return []
+    const pageMap = (__nextra_internal__.pageMap as PageMapItem[]).find(
+      (pageItem): pageItem is Folder => pageItem.kind === 'Folder' && pageItem.name === locale
+    )!.children
+    pageMap.find((pageItem): pageItem is MetaJsonFile => pageItem.kind === 'Meta')!.data.index = translate(
+      translations,
+      locale,
+      'index.title'
+    )
+    return pageMap
+  }, [__nextra_internal__.pageMap, hasMDXPage, locale])
```

Comment:
This whole thing seems pretty hacky. 😅

Reply:
it is hacky due `[locale]/index.mdx` file if we'll refactor to `en/index.mdx`, `fr/index.mdx` the whole mentioned code will be a lot simplified (because normally TOC headings are parsed but here I inject these headings only for this `index` page)

Reply:
custom styles can be extracted as React components but headings need to exist as raw markdown headings

```md
---
title: My title
---

## Foo bar

<SomeSections />

### Qux

<SomeLogos />

```

Reply:
Ok, good idea.

Reply:
I will create an issue for 👆, will rename `theme` folder to `nextra-theme` and we are ready to merge it!

The next one is Remote MDX PR!